library/src/test/java/kwasm/format/text/TableTypeTest.kt

```diff
@@ -0,0 +1,161 @@
+        assertThat(exception).hasMessageThat().contains("Too few arguments")
+    }
+
+    @Test
+    fun parseSingleNumber_withoutNonFuncrefLiteral_throwsParseException() {
+        val expectedMin = 123456.toUInt()
+        val tokens = tokenizer.tokenize("$expectedMin notFuncref", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Expected 'funcref'")
+    }
+
+    @Test
+    fun parseSingleNumber_setMinToCorrectValue() {
+        val expectedMin = 123456.toUInt()
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseMaxVal_setMinToMaxVal() {
+        val expectedMin = UInt.MAX_VALUE
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseMinVal_setMinToMinVal() {
+        val expectedMin = UInt.MIN_VALUE
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseNegativeMin_throwsParseExceptionWithNegativeNumberMessage() {
+        val tokens = tokenizer.tokenize("-123456", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Expected integer literal")
+    }
+
+    @Test
+    fun parseABitLargerThanMaxVal_throwsParseExceptionWithValueOverflowMessage() {
+        val tokens = tokenizer.tokenize("4294967296", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Illegal value")
+    }
+
+    @Test
+    fun parseALotLargerThanMaxVal_throwsParseExceptionWithValueOverflowMessage() {
+        val tokens = tokenizer.tokenize("100000000000", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Illegal value")
+    }
```

Comment:
I think these tests are fine to omit. They're handled by the Limits-parsing test cases, right?